internal/config/sshSetup.go

```diff
@@ -0,0 +1,142 @@
+package config
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+)
+
+func createKeyPair(privateKeyPath, publicKeyPath string) error {
```

Comment:
About this function signature. Since you're assuming the public key name to be `privateKeyName + ".pub"` everywhere else, why not have just one input (`keyPath`) and append `.pub` to get the public key path?

Reply:
I left it this way because we're either way defining both in the `SetAuth()` function. And in case we might use the function elsewhere, then we could just export the function without making any more changes.